File: semestru 4/MotoProject/Networking/src/main/java/network/jsonprotocol/JsonProtocolUtils.java
package network.jsonprotocol;

import model.Player;
import model.Race;
import model.User;
import network.dto.DTOUtils;
import network.dto.PlayerDTO;
import network.dto.TeamDTO;
import network.dto.UserDTO;

public class JsonProtocolUtils {



    public static Response createOkResponse(){
        Response resp=new Response();
        resp.setType(ResponseType.OK);
        return resp;
    }

    public static Response createErrorResponse(String errorMessage){
        Response resp=new Response();
        resp.setType(ResponseType.ERROR);
        resp.setErrorMessage(errorMessage);
        return resp;
    }

    public static Response createGetRacesResponse(Race[] races){
        Response resp = new Response();
        resp.setType(ResponseType.GET_RACES);
        resp.setRaces(DTOUtils.getDTO(races));
        return resp;
    }

    public static Response createGetPlayerResponse(Player[] players){
        Response resp = new Response();
        resp.setType(ResponseType.GET_TEAM_PLAYERS);
        resp.setPlayers(DTOUtils.getDTO(players));
        return resp;
    }
    public static Response createGetTeamResponse(TeamDTO team){
        Response resp = new Response();
        resp.setType(ResponseType.GET_TEAM);
        resp.setTeam(team);
        return resp;
    }
    public static Request createLoginRequest(User user){
        Request req=new Request();
        req.setType(RequestType.LOGIN);
        req.setUser(DTOUtils.getDTO(user));
        return req;
    }

    public static Request createLogoutRequest(User user){
        Request req=new Request();
        req.setType(RequestType.LOGOUT);
        req.setUser(DTOUtils.getDTO(user));
        return req;
    }
    public static Request createAddPlayerRequest(Player player,String raceId){
        Request req=new Request();
        req.setType(RequestType.ADD_PLAYER);
        req.setPlayer(DTOUtils.getDTO(player));
        req.setRaceId(raceId);
        return req;
    }
    public static Request createGetRacesRequest(){
        Request req=new Request();
        req.setType(RequestType.GET_RACES);
        return req;
    }

    public static Request createGetTeamRequest(String teamName){
        Request req=new Request();
        req.setType(RequestType.GET_TEAM);
        req.setTeamName(teamName);
        return req;
    }
    public static Request createGetTeamPlayersRequest(String raceId,String teamId){
        Request req=new Request();
        req.setType(RequestType.GET_TEAM_PLAYERS);
        req.setTeamId(teamId);
        req.setRaceId(raceId);
        return req;
    }


    public static Response createPlayerAddedResponse(PlayerDTO dto) {
        Response resp = new Response();
        resp.setType(ResponseType.NEW_PLAYER);
        resp.setNewPlayer(dto);
        return resp;
    }
}

File: semestru 4/MotoProjectGrpcCs/networking/jsonprotocol/JsonProtocolUtils.cs
using moto.model;
using moto.networking.dto;

namespace moto.networking.jsonprotocol;

public class JsonProtocolUtils
{
    public static Response CreateOkResponse()
    {
        return new Response { Type = ResponseType.OK };
    }

    public static Response CreateErrorResponse(string errorMessage)
    {
        return new Response { Type = ResponseType.ERROR, ErrorMessage = errorMessage };
    }

    public static Response CreateGetRacesResponse(Race[] races)
    {
        return new Response { Type = ResponseType.GET_RACES, Races = DTOUtils.GetDTO(races) };
    }

    public static Response CreateGetPlayerResponse(Player[] players)
    {
        return new Response { Type = ResponseType.GET_TEAM_PLAYERS, Players = DTOUtils.GetDTO(players) };
    }

    public static Response CreateGetTeamResponse(TeamDTO team)
    {
        return new Response { Type = ResponseType.GET_TEAM, Team = team };
    }

    public static Request CreateLoginRequest(User user)
    {
        return new Request { Type = RequestType.LOGIN, User = DTOUtils.GetDTO(user) };
    }

    public static Request CreateLogoutRequest(User user)
    {
        return new Request { Type = RequestType.LOGOUT, User = DTOUtils.GetDTO(user) };
    }

    public static Request CreateAddPlayerRequest(Player player, string raceId)
    {
        return new Request { Type = RequestType.ADD_PLAYER, Player = DTOUtils.GetDTO(player), RaceId = raceId };
    }

    public static Request CreateGetRacesRequest()
    {
        return new Request { Type = RequestType.GET_RACES };
    }

    public static Request CreateGetTeamRequest(string teamName)
    {
        return new Request { Type = RequestType.GET_TEAM, TeamName = teamName };
    }

    public static Request CreateGetTeamPlayersRequest(string raceId, string teamId)
    {
        return new Request { Type = RequestType.GET_TEAM_PLAYERS, RaceId = raceId, TeamId = teamId };
    }

    public static Response CreatePlayerAddedResponse(PlayerDTO dto)
    {
        return new Response { Type = ResponseType.NEW_PLAYER, NewPlayer = dto };
    }
}